--- /usr/share/flash-kernel/db/all.db.orig	2014-10-01 08:54:52.000000000 +0000
+++ /usr/share/flash-kernel/db/all.db	2014-10-01 09:03:36.000000000 +0000
@@ -352,6 +352,17 @@
 Required-Packages: u-boot-tools
 Bootloader-sets-root: no
 
+Machine: Insignal Arndale evaluation board based on EXYNOS5250
+Kernel-Flavors: generic
+U-Boot-Kernel-Address: 0x40008000
+U-Boot-Initrd-Address: 0x0
+DTB-Id: exynos5250-arndale.dtb
+DTB-Kernel-Version: 3.16
+Boot-Kernel-Path: /boot/uImage
+Boot-Initrd-Path: /boot/uInitrd
+Required-Packages: u-boot-tools
+Bootloader-sets-root: no
+
 Machine: Seagate FreeAgent DockStar
 Kernel-Flavors: kirkwood
 U-Boot-Kernel-Address: 0x00008000
